src/modules/fw_pos_control/FixedwingPositionControl.cpp

```diff
@@ -2356,18 +2356,48 @@ FixedwingPositionControl::control_manual_position(const float control_interval,
 	attitude_setpoint.copyTo(_att_sp.q_d);
 }
 
-void FixedwingPositionControl::control_backtransition(const float control_interval, const Vector2f &ground_speed,
-		const position_setpoint_s &pos_sp_curr)
+void FixedwingPositionControl::control_backtransition_heading_hold()
 {
-	const bool is_low_height = checkLowHeightConditions();
+	if (!PX4_ISFINITE(_backtrans_heading)) {
+		_backtrans_heading = _local_pos.heading;
+	}
 
-	float target_airspeed = adapt_airspeed_setpoint(control_interval, pos_sp_curr.cruising_speed,
-				_performance_model.getMinimumCalibratedAirspeed(getLoadFactor()), ground_speed);
+	float true_airspeed = _airspeed_eas * _eas2tas;
+
+	if (!_airspeed_valid) {
+		true_airspeed = _performance_model.getCalibratedTrimAirspeed() * _eas2tas;
+	}
+
+	Vector2f airspeed_vector = Vector2f(cosf(_local_pos.heading), sinf(_local_pos.heading)) * true_airspeed;
```

Comment:
@RomanBapst have you checked in simulation how well this works with a high crosswind backtransition? As there the course is not equal tot he heading, so this assumption here gets stretched...

Reply:
@sfuhrer I will run a simulation with high wind and post results here.